x/warp/predicate_test.go

```diff
@@ -255,6 +255,80 @@ func TestWarpNilProposerCtx(t *testing.T) {
 	test.Run(t)
 }
 
+func TestWarpMessageFromPrimaryNetwork(t *testing.T) {
+	numKeys := 10
+	cChainID := ids.GenerateTestID()
+	unsignedMsg, err := avalancheWarp.NewUnsignedMessage(networkID, cChainID, []byte{1, 2, 3})
+	if err != nil {
+		t.Fatal(err)
+	}
+	getValidatorsOutput := make(map[ids.NodeID]*validators.GetValidatorOutput)
+	blsSignatures := make([]*bls.Signature, 0, numKeys)
+	for i := 0; i < numKeys; i++ {
+		validatorOutput := &validators.GetValidatorOutput{
+			NodeID:    testVdrs[i].nodeID,
+			Weight:    20,
+			PublicKey: testVdrs[i].vdr.PublicKey,
+		}
+		getValidatorsOutput[testVdrs[i].nodeID] = validatorOutput
+		blsSignatures = append(blsSignatures, bls.Sign(testVdrs[i].sk, unsignedMsg.Bytes()))
+	}
+	aggregateSignature, err := bls.AggregateSignatures(blsSignatures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	bitSet := set.NewBits()
+	for i := 0; i < numKeys; i++ {
+		bitSet.Add(i)
+	}
+	warpSignature := &avalancheWarp.BitSetSignature{
+		Signers: bitSet.Bytes(),
+	}
+	copy(warpSignature.Signature[:], bls.SignatureToBytes(aggregateSignature))
+	warpMsg, err := avalancheWarp.NewMessage(unsignedMsg, warpSignature)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	predicateBytes := predicateutils.PackPredicate(warpMsg.Bytes())
+
+	snowCtx := snow.DefaultContextTest()
+	snowCtx.SubnetID = ids.GenerateTestID()
+	snowCtx.ChainID = ids.GenerateTestID()
+	snowCtx.CChainID = cChainID
+	snowCtx.NetworkID = networkID
```

Comment:
Updating the line below to return `constants.PrimaryNetworkID` instead of `ids.Empty`. The `SubnetID` declared here is meant to be a non-PrimaryNetwork SubnetID since it's receiving from the PrimaryNetwork in this test.